submodules/telemetry/__init__.py

```diff
@@ -1,133 +1,145 @@
-                else:
-                    squishedpackets += log_stack.pop().to_string()
-            squishedpackets = base64.b64encode(squishedpackets.encode('ascii'))
-            radio_output.send(squishedpackets, radio)
-            squishedpackets = ""
-
-
-def clear_buffers() -> None:
-    """
-    Clear the telemetry buffers - clearing general_queue, the log, and error stacks.
-    :return: None
-    """
-    global packet_lock, log_stack, err_stack, general_queue
-    with packet_lock:
-        general_queue.clear()
-        log_stack.clear()
-        err_stack.clear()
-
-
-def decide() -> None:
-    """
-    A thread method to constantly check general_queue for messages and process them if there are any.
-    :return: None
-    """
-    global packet_lock, err_stack, log_stack, general_queue
-    while True:
-        if len(general_queue) != 0:
-            with packet_lock:
-                message = general_queue.popleft()
-                if type(message) is str and message[0] == ';':
-                    command_ingest.enqueue(message)
-                    #print("Running command_ingest.enqueue(" + message + ")")
-                elif type(message) is error.Error:
-                    err_stack.append(message)
-                elif type(message) is log.Log:
-                    log_stack.append(message)
-                else:  # Shouldn't execute (enqueue() should catch it) but here just in case
-                    logger.error("Message prefix invalid.")
-        sleep(1)
-
-
-def start() -> None:
-    """
-    Starts the telemetry send thread
-    :return None
-    """
-    global err_stack, log_stack, general_queue, packet_lock
-    general_queue = collections.deque()  # initialize global variables
-    log_stack = collections.deque()
-    err_stack = collections.deque()
-    packet_lock = Lock()
-
-    threadDecide = ThreadHandler(target=partial(decide), name="telemetry-decide")  # start telemetry 'decide' thread
-    threadDecide.start()
-
-
-def enter_normal_mode() -> None:
-    """
-    Enter normal mode.
-    :return: None
-    """
-    global state
-    state = Mode.NORMAL
-
-
-def enter_low_power_mode() -> None:
-    """
-    Enter low power mode.
-    :return: None
-    """
-    global state
-    state = Mode.LOW_POWER
-
-
-def enter_emergency_mode() -> None:
-    """
-    Enter emergency mode.
-    :return: None
-    """
-    global state
-    state = Mode.EMERGENCY
+class Telemetry:
+    def __init__(self, config):
+        """
+        Constructor method. Initializes variables
+        :param config: Config variable passed in from core.
+        """
+        self.config = config
+        self.modules = {}
+        self.general_queue = collections.deque()  # initialize global variables
+        self.log_stack = collections.deque()
+        self.err_stack = collections.deque()
+        self.packet_lock = Lock()
+        self.logger = logging.getLogger("TELEMETRY")
+
```

Comment:
add `self.processes`

```
self.processes = {
 "telemetry-decide": ThreadHandler(..._)
}
```

Reply:
Fixed in 149ce11

Reply:
dont create separate objects for threadhandlers. Instantiate the `ThreadHandler` objects inline in `self.processes`